tests/microbenchmarks/all_reduce_rmsnorm_compare.py

```
@@ -0,0 +1,459 @@
+    dist = Distributed.get(mapping)
+
+    torch_dtype = str_dtype_to_torch(dtype)
+    element_size = torch.tensor([], dtype=torch_dtype).element_size()
+    max_elements = max(num_tokens * hidden_size for num_tokens, hidden_size in shapes)
+    strategies = _init_userbuffers_if_needed(strategies, mapping, max_elements, element_size)
+
+    allreduce_instances = {}
+    for strategy in strategies:
+        try:
+            allreduce_instances[strategy] = AllReduce(mapping=mapping, strategy=strategy, dtype=torch_dtype)
+        except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)